src/resolvers/workspace.js

```diff
@@ -319,6 +320,62 @@ module.exports = {
       return true;
     },
 
+    /**
+     * Mutation in order to leave workspace
+     * @param {ResolverObj} _obj - object that contains the result returned from the resolver on the parent field
+     * @param {string} workspaceId - id of the workspace where the user should be removed
+     * @param {UserInContext} user - current authorized user {@see ../index.js}
+     * @param {ContextFactories} factories - factories for working with models
+     * @return {Promise<boolean>} - true if operation is successful
+     */
+    async deleteWorkspace(_obj, { workspaceId }, { user, factories }) {
+      const workspaceModel = await factories.workspacesFactory.findById(workspaceId);
+
+      if (!workspaceModel) {
+        throw new UserInputError('There is no workspace with provided id');
+      }
+
+      const memberInfo = await workspaceModel.getMemberInfo(user.id);
+
+      /**
+       * Return if user is not Admin.
+       */
+      if (!memberInfo.isAdmin) {
+        return false;
+      }
+
+      const membersInfo = (await workspaceModel.getMembers());
+
+      for (const member of membersInfo) {
+        const userModel = await factories.usersFactory.findById(member.userId.toString());
+
+        await userModel.markWorkspaceAsRemoved(workspaceId.toString());
+      }
+
+      const projectToWorkspace = new ProjectToWorkspace(workspaceId.toString());
+
+      const projectsInfo = await projectToWorkspace.getProjects();
+
+      if (projectsInfo.length) {
+        for (const project of projectsInfo) {
+          /**
+           * Remove project events
+           */
+          await new EventsFactory(project._id).remove();
+          /**
+           * Remove project
+           */
+          const projectModel = await factories.projectsFactory.findById(project.id.toString());
+
+          await projectModel.markProjectAsRemoved();
```

Comment:
oh, also need to update collector's logic not to get projects with `isRemoved` flag

Reply:
I guess no need since we are taking user's only workspace which doesn't contain `isRemoved` flag which contain the projects which doesn't contain `isRemoved` flag eventually